### PR TITLE
[pairing] Refactor remote_pairing to use cmdbase

### PR DESCRIPTION
**Pull request by chme**
@ejurgensen This requires a bit more work, but a short test was successful. I am opening the PR early to get feedback, if this change is acceptable. The goal is to be able to call the pairing request with a synchronous command from the JSON API and return a meaningful response (success or failure) to e. g. the web interface.

### CODE REVIEW

**Comment by ejurgensen** [comment]
I think this looks very good. When you think it is ready you are welcome to merge it in.

**Comment by chme** [comment]
This is now ready to be merged. @ejurgensen  if you have time, would be nice, if you can take another look. I am not too happy about the new error constants, but i want to distinguish internal errors from bad user input. If there are no remarks, i will probably merge in the next days.

File: src/remote_pairing.c [hunk 1]
```diff
@@ -779,6 +703,71 @@ remote_pairing_get_name(void)
...
+
+  cmdarg = strdup(arglist[0]);
+
+  DPRINTF(E_LOG, L_REMOTE, "Kickoff pairing with pin '%s'\n", arglist[0]);
+
+  commands_exec_async(cmdbase, pairing_pair, cmdarg);
+}
+
+int
+remote_pairing_pair(char *pin)
```
Comment (ejurgensen) [inline src/remote_pairing.c]: Is it possible to have const char *pin?

**Comment by chme** [comment]
> I did notice one thing that looked odd, but maybe I am reading it wrong?

Good catch, it should return 1 for a successful request for ipv6. Fixed it, but if i read correctly, IPv6 is currently not supported, at least that is what the comment in `remote_pairing_init` says. Also made the pin parameter a `const char`.

**Comment by ejurgensen** [comment]
> it should return 1 for a successful request for ipv6

I suggest you let do_pairing() use the standard -1/0 return values (and of course change the test in line 730).

I don't know if Remote these days supports ipv6, but even if it doesn't I suggest leaving that part as is it - some day in the future I expect Remote it will be supported.

**Comment by chme** [comment]
Done

**Comment by ejurgensen** [comment]
Merged, thanks

**Pull request merged**